hpx/lcos/detail/promise_base.hpp

```diff
@@ -316,6 +316,8 @@ namespace lcos {
                   : ptr_(o.ptr_.release(), &wrapping_deleter)
                 {}
 
+                keep_alive& operator=(keep_alive&& o) = default;
+
```

Comment:
Should we add a default copy assignment operator as well?

Reply:
No. This thing is obviously not copyable

Reply:
Hmm, that's not really obvious to me.  If the intention is to inhibit copying, add a `=delete` for the corresponding functions.